File: angular-16-jwt-auth-master/src/app/components/admin/token-management/token.ts
import { TokenState } from "./tokenState.enum";

export interface Token{
    id?: string;
    tokenNum?: number;
    selectedDay?: string;
    state?: TokenState;
    reservedByID?: number;
    reservedByName?: string;
}

export class TokenImplementation implements Token {
    constructor(
      public tokenNum?: number,
      public selectedDay?: string,
      public reservedByID?: number
    ) {}
  }

File: TokenManager/src/main/java/com/hSenid/demo/models/Token.java
package com.hSenid.demo.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document (collection = "tokens")
public class Token {

    @Transient
    public static String SEQUENCE_NAME = "token_sequence";
    @Id
    private String id;
    @Indexed(unique = true)
    private int tokenNum;
    @Field("selectedDay")
    private LocalDate selectedDay;
    private TokenState state;
    private int reservedByID;
    private String reservedByName;

}
